### PR TITLE
Added iOS template for the example applications

### PR DESCRIPTION
**Pull request by borzun**
- added iOS template for automating the generating of new example applications;
- configured dataservice-write example for iOS based on that template;

Resolves: OLPEDGE-287

Signed-off-by: Bohdan Kurylovych <ext-bohdan.kurylovych@here.com>

### CODE REVIEW

File: docs/dataservice-write-example.md [hunk 1]
```diff
@@ -77,6 +77,28 @@ After installing and running the `dataservice_write_example` apk, you should see
 
 Note, that you can run `CMake` command directly from `<olp-edge-sdk-root>/examples/dataservice-write/` folder if you have already built and installed OLP C++ EDGE SDK libraries for Android. Make sure that you pass the correct path to `LevelDB` library and provided the correct `EDGE_SDK_NETWORK_PROTOCOL_JAR` parameter in the `CMake` command invoked by the `build/examples/dataservice-write/android/app/build.gradle` script.
 
+## Building and running on iOS
+
+This example shows how to integrate and use the OLP C++ EDGE SDK in a basic iOS application written in Objective-C language.
```
Comment (andescu) [inline docs/dataservice-write-example.md]: I just notice that we use here `OLP C++ EDGE SDK` but in README.md `HERE OLP Edge SDK`
We need to align. Please change all occurrences to `HERE OLP Edge SDK`

File: docs/dataservice-write-example.md [hunk 2]
```diff
@@ -77,6 +77,28 @@ After installing and running the `dataservice_write_example` apk, you should see
...
+
```
Comment (andescu) [inline docs/dataservice-write-example.md]: Please follow same structure as for Android.
Means:
### Prerequisities
### Build...
### Assemble... & Additional notes [IF NEEDED]

File: docs/dataservice-write-example.md [hunk 3]
```diff
@@ -77,6 +77,28 @@ After installing and running the `dataservice_write_example` apk, you should see
...
+Configure the OLP C++ EDGE SDK with `EDGE_SDK_BUILD_EXAMPLES` variable set to `ON`, disable tests via `EDGE_SDK_ENABLE_TESTING` variable, and specify path to the iOS toolchain file shipped with SDK (located in `<olp-edge-sdk-root>/cmake/toolchains/iOS.cmake`):
+
```
Comment (andescu) [inline docs/dataservice-write-example.md]:
```suggestion
First, before building, you need to configure the HERE OLP Edge SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, disable tests with `EDGE_SDK_ENABLE_TESTING` set to `OFF`, and specify the path to the iOS toolchain file shipped together with the SDK and located under `${PROJECT_SOURCE_DIR}/cmake/toolchains/iOS.cmake`:
```

File: docs/dataservice-write-example.md [hunk 4]
```diff
@@ -77,6 +77,28 @@ After installing and running the `dataservice_write_example` apk, you should see
...
+```bash
+mkdir build && cd build
+cmake .. -GXcode  -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/iOS.cmake -DPLATFORM=iphoneos -DEDGE_SDK_BUILD_EXAMPLES=ON -DEDGE_SDK_ENABLE_TESTING=OFF
+```
+
+(note that in order to configure the OLP C++ EDGE SDK for a simulator, set `SIMULATOR` variable to `ON`)
```
Comment (andescu) [inline docs/dataservice-write-example.md]: Remove braces.

Comment (andescu) [inline docs/dataservice-write-example.md]:
```suggestion
Note that in order to configure the HERE OLP Edge SDK C++ for a simulator, you need to set the `SIMULATOR` variable to `ON`.
```

File: docs/dataservice-write-example.md [hunk 5]
```diff
@@ -77,6 +77,28 @@ After installing and running the `dataservice_write_example` apk, you should see
...
+
+Now open the generated `XCode` project:
+```bash
+open olp-cpp-sdk.xcodeproj
+```
+
+Select the `dataservice-write-example` scheme from the schemes list in `XCode` and specify correct your team identifier for the `dataservice-write-example` target.
```
Comment (andescu) [inline docs/dataservice-write-example.md]:
```suggestion
Select the `dataservice-write-example` scheme from the schemata list in `XCode` and specify your correct team identifier for the `dataservice-write-example` target.
```

File: docs/dataservice-write-example.md [hunk 6]
```diff
@@ -77,6 +77,28 @@ After installing and running the `dataservice_write_example` apk, you should see
...
+
+Once everything is correctly setup, build and run the example application on your device and you should see the `Publish Successful` message in the main UI screen. If you encountered an error (`Publish Failed` message), please check the device's logs for the detailed description of the error message.
```
Comment (andescu) [inline docs/dataservice-write-example.md]:
```suggestion
Once everything is correctly set up, build and run the example application on your device and you should see the `Publish Successful` message in the main UI screen. If you encountered an error message, e.g. `Publish Failed`, please check the device's logs for the detailed description of the error.
```

File: examples/helpers/ios/CMakeLists.txt.in [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+
+cmake_minimum_required(VERSION 3.5)
+
+if (NOT IOS)
+    message(FATAL_ERROR "Unsupported architecture!")
```
Comment (andescu) [inline examples/helpers/ios/CMakeLists.txt.in]:
```suggestion
    message(FATAL_ERROR "Unsupported platform!")
```

Comment (borzun) [reply examples/helpers/ios/CMakeLists.txt.in]: Done

File: docs/dataservice-write-example.md (hunk 6 above)
Comment (borzun) [reply docs/dataservice-write-example.md]: Done

File: docs/dataservice-write-example.md (hunk 5 above)
Comment (borzun) [reply docs/dataservice-write-example.md]: I changed a bit this sentence. Btw, what wrong with schemes list? Schemata is plural for schema, but XCode uses **scheme** notion:  https://developer.apple.com/library/archive/featuredarticles/XcodeConcepts/Concept-Schemes.html

File: docs/dataservice-write-example.md (hunk 4 above)
Comment (borzun) [reply docs/dataservice-write-example.md]: Done

File: docs/dataservice-write-example.md (hunk 3 above)
Comment (borzun) [reply docs/dataservice-write-example.md]: Done

File: docs/dataservice-write-example.md (hunk 2 above)
Comment (borzun) [reply docs/dataservice-write-example.md]: Done

File: docs/dataservice-write-example.md (hunk 1 above)
Comment (borzun) [reply docs/dataservice-write-example.md]: Done

File: docs/dataservice-write-example.md (hunk 5 above)
Comment (andescu) [reply docs/dataservice-write-example.md]: I thought it was meant to be schema.
Ignore my comment then.

**Pull request merged**